coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -194,5 +203,23 @@ private RexCall adjustInconsistentTypesToEqualityOperator(RexCall call) {
       }
       return call;
     }
+
+    private RexNode convertMapValueConstructor(RexBuilder rexBuilder, RexCall call) {
+      List<RexNode> sourceOperands = call.getOperands();
+      final List<RexNode> results = new ArrayList<>();
+      // Even numbers are keys
+      List<RexNode> keys = new ArrayList<>();
+      for (int i = 0; i < sourceOperands.size(); i += 2) {
+        keys.add(sourceOperands.get(i));
+      }
+      results.add(rexBuilder.makeCall(SqlStdOperatorTable.ARRAY_VALUE_CONSTRUCTOR, keys));
+      // Odd numbers are values
+      List<RexNode> values = new ArrayList<>();
+      for (int i = 1; i < sourceOperands.size(); i += 2) {
+        values.add(sourceOperands.get(i));
```

Comment:
If value is also a map, we failed to convert it.
For example, if the view definition is:
```
SELECT MAP('key1', MAP('a', 'b', 'c', 'd'), 'key2', MAP('a', 'b', 'c', 'd')) AS col_map
```
it would only be converted to
```
SELECT MAP (ARRAY['key1', 'key2'], ARRAY[MAP ('a', 'b', 'c', 'd'), MAP ('a', 'b', 'c', 'd')]) AS "col_map"
```
However, the result should be
```
SELECT MAP (ARRAY['key1', 'key2'], ARRAY[MAP (ARRAY['a', 'c'], ARRAY['b', 'd']), MAP (ARRAY['a', 'c'], ARRAY['b', 'd'])]) AS "col_map"
```
I think we can change this part to be:
```
if (sourceOperands.get(i) instanceof RexCall) {
  values.add(visitCall((RexCall) sourceOperands.get(i)));
} else {
  values.add(sourceOperands.get(i));
}
```
and add corresponding unit tests.

Reply:
I agree with you and will try to fix it.